examples/tut-6-rpc/client-py/index.py

```diff
@@ -0,0 +1,108 @@
+import json
+import os
+import random
+import requests
+from concurrent.futures import ALL_COMPLETED, ThreadPoolExecutor, wait
+from time               import sleep
+
+
+def log(t):
+    print(t, flush=True)
+
+
+def mk_rpc(proto, host, port):
+    def rpc(m, *args):
+        lab = 'RPC %s %s' % (m, json.dumps([*args]))
+        log(lab)
+        ans = requests.post('%s://%s:%s%s' % (proto, host, port, m),
+                            json=[*args])
+        log('%s ==> %s' % (lab, json.dumps(ans.json())))
+        return ans.json()
+
+    def rpc_callbacks(m, arg, cbacks):
+        vals  = {k: v    for k, v in cbacks.items() if not callable(v)}
+        meths = {k: True for k, v in cbacks.items() if     callable(v)}  # noqa
+        p     = rpc(m, arg, vals, meths)
+
+        while True:
+            if p['t'] == 'Done':
+                return p
+
+            elif p['t'] == 'Kont':
+                cback = cbacks[p['m']]
+                ans   = cback(*p['args'])
+                p     = rpc('/kont', p['kid'], ans)
+
+            else:
+                # TODO fix swallowed exceptions
+                raise Exception('Illegal callback return: %s' % json.dumps(p))
+
+    return rpc, rpc_callbacks
+
+
+def main():
+    log('I am the client')
+    sleep(3)  # TODO wait until server becomes available to fulfill requests
+    host = os.environ['REACH_RPC_SERVER']
+    port = os.environ['REACH_RPC_PORT']
+
+    rpc, rpc_callbacks = mk_rpc('http', host, port)
+
+    starting_balance = rpc('/stdlib/parseCurrency', 10)
+    acc_alice        = rpc('/stdlib/newTestAccount', starting_balance)
+    acc_bob          = rpc('/stdlib/newTestAccount', starting_balance)
+
+    fmt          = lambda x: rpc('/stdlib/formatCurrency', x, 4)
+    get_balance  = lambda w: fmt(rpc('/stdlib/balanceOf', w))
```

Comment:
Would "normal Python people" write these as functions some other way?

Reply:
This "isn't Pythonic", no... They want you to use several lines of the form:
```python
def fmt(x):
    return rpc('/stdlib/formatCurrency', x, 4)

# "PEP8" has different rules about how many lines must separate function definitions based on scope

def whatever(foo):
    pass
```

Better to appeal to familiarity than concision?

Reply:
Yea, I think we should go with familiarity. I don't want Python people to think they NEED to use `lambda`. Similarly, is it the case that the callbacks would "just work" on an object?

Reply:
What I mean by an object is... see the next comment